app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -202,6 +204,13 @@ def reset_katello_status
         true
       end
 
+      def ensure_no_special_chars
+        hostname = self.name
+        unless hostname.scan(/[!@#$%^&*()_+{}\[\]:;'"\/\\?><,]/).empty?
```

Comment:
This regex seems to allow capital letters and spaces.

I looked on the internet and found 
```suggestion
        unless hostname.match?(/(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{1,63}(?<!-)\.)+[a-zA-Z]{2,63}$)/)
```

https://stackoverflow.com/questions/11809631/fully-qualified-domain-name-validation

Thoughts?